### PR TITLE
backlog task: add eslint and prettier to the project

### PR DESCRIPTION
**Pull request by orama254**
## Fixes Issue

configure eslint and prettier for project wide code uniformity

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [X] My change requires changes to the documentation. **(an extra step to run lint and prettier before commits)**
- [ ] I have updated the documentation accordingly. **(Will do after this PR)**
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,20 +7,22 @@
     "build": "next build",
     "start": "next start",
     "lint": "next lint",
-    "test": "playwright test"
+    "test": "playwright test",
+    "prettier": "prettier --write ."
```
Comment (antosan) [inline package.json:11]: I suggest we only format the files which are tracked in the repository by changing this to `"prettier --write --ignore-path .gitignore ."`.

File: src/pages/index.page.jsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import Head from "next/head";
-import Banner from "../components/Banner";
-import Image from "next/image";
-import logo from "../../public/reactdevske.svg";
+import Head from 'next/head';
+// import Banner from "../components/Banner";   // commented out due to lint error during lint configuration
```
Comment (antosan) [inline src/pages/index.page.jsx:2]: We can remove the comment as well since the component is not used in this file.

**Review by antosan — COMMENTED** [review]
Nice job @orama254 🎉 I have some comments:

1. We should remove the `.eslintrc.json` line from the `.gitignore` so that git can track the `.eslintrc.json` file. After removing it, run `npm run lint`, choose strict and include the generated `.eslintrc.json` file in the committed files.

	Also, `eslint` rules usually conflict with `prettier` rules and in order to prevent this:

	- Run `npm install --save-dev eslint-config-prettier`
	- Update `.eslintrc.json` to:

	```json
	{
  		"extends": ["next/core-web-vitals", "prettier"]
	}
	```
	
2. Whenever I open a `.jsx` file that does not include `import React from 'react';` in VSCode, e.g. `src/pages/index.page.jsx`, I get a lot of warnings. Do you have the same?

image

File: package.json (hunk 1 above)
Comment (orama254) [reply package.json]: made a commit with the latest changes based on these suggestions

File: src/pages/index.page.jsx (hunk 2 above)
Comment (orama254) [reply src/pages/index.page.jsx]: done ✅

**Comment by orama254** [comment]
> Nice job @orama254 🎉 I have some comments:
> 
> 1. We should remove the `.eslintrc.json` line from the `.gitignore` so that git can track the `.eslintrc.json` file. After removing it, run `npm run lint`, choose strict and include the generated `.eslintrc.json` file in the committed files.
>    Also, `eslint` rules usually conflict with `prettier` rules and in order to prevent this:
>    
>    * Run `npm install --save-dev eslint-config-prettier`
>    * Update `.eslintrc.json` to:
>    
>    ```json
>    {
>    	"extends": ["next/core-web-vitals", "prettier"]
>    }
>    ```
> 2. Whenever I open a `.jsx` file that does not include `import React from 'react';` in VSCode, e.g. `src/pages/index.page.jsx`, I get a lot of warnings. Do you have the same?
> 
> image

@antosan for the second point there is a line on the lint config that should disable this. Could you try it out now that the file isn't ignored inside git.

 ```json
     {
   	  "globals": {
                "React": "readonly"
            },
     }
 ```

**Comment by antosan** [comment]
> @antosan for the second point there is a line on the lint config that should disable this. Could you try it out now that the file isn't ignored inside git.

This didn't fix it for me, probably it is just my editor misbehaving.

**Review by antosan — APPROVED** [review]
LGTM 👍

**Pull request merged**